src/relations/mysql_provider.py

```diff
@@ -212,16 +212,25 @@ def _on_database_requested(self, event: DatabaseRequestedEvent):
             self.database.set_endpoints(relation_id, rw_endpoints)
             self.database.set_version(relation_id, db_version)
             self.database.set_read_only_endpoints(relation_id, ro_endpoints)
-            # TODO:
-            # add setup of tls, tls_ca and status
-            self.charm._mysql.create_application_database_and_scoped_user(
-                db_name, db_user, db_pass, "%"
-            )
 
             if "mysqlrouter" in extra_user_roles:
+                self.charm._mysql.create_application_database_and_scoped_user(
+                    db_name,
+                    db_user,
+                    db_pass,
+                    "%",
+                    # MySQL Router charm does not need a new database
+                    create_database=False,
+                )
                 self.charm._mysql.grant_privileges_to_user(
                     db_user, "%", ["ALL PRIVILEGES"], with_grant_option=True
                 )
+            else:
+                # TODO:
+                # add setup of tls, tls_ca and status
```

Comment:
JFMI, is it still topical? mysql-router-k8s supports TLS since the last sprint AFAIK, mysql-router is subordinate == no TLS.